src/merge_iterator.rs

```diff
@@ -444,6 +477,34 @@ mod tests {
         .await;
     }
 
+    #[tokio::test]
```

Comment:
we should add test cases where a key appears in both/multiple iterators for twomerge/merge iterators respectively, with varying ordering of the sequence numbers across the inner iterators